### ISSUE TITLE
linter: oxlint failed to parse rule import/no-unassigned-import

### ISSUE THREAD

**Issue by SaLapus** [oxc-project/oxc#19288]
### What version of Oxlint are you using?

1.46.0

### What command did you run?

`yarn oxlint --config='oxlint.config.ts'`

### What does your `.oxlintrc.json` config file look like?

```ts
import { defineConfig } from "oxlint"

export default defineConfig({
  plugins: ["import", "promise"],
  categories: {
    correctness: "error",
    suspicious: "warn",
    perf: "warn",
  },
  rules: {
    "import/no-unassigned-import": [{ allow: ["**/*.css"] }],
  },
})
```

### What happened?

[Documentation for this rule](https://oxc.rs/docs/guide/usage/linter/rules/import/no-unassigned-import.html) says it accepts a configuration object but in reality oxlint fails trying to parse rule with error "expected a string or a number, but got Object {"allow": Array [String("**/*.css")]}"
TypeScript type-checking also gives error, because 'allow' does not exist in type 'unknown[]'
`type DummyRule = AllowWarnDeny | unknown[];`

**Comment by camc314** [comment]
you need to specify the severity of the rule first

```
    "import/no-unassigned-import":  ["error", { allow: ["**/*.css"] }],

**Comment by SaLapus** [comment]
@camc314 Thank you.
So its meant to be always a string or tuple `[severity, options]` with only one configuration object of options?
Are there any obstacles to changing this type?
From
```json
{
  "DummyRule": {
    "anyOf": [
      {
        "$ref": "#/definitions/AllowWarnDeny"
      },
      {
        "type": "array",
        "items": true
      }
    ]
  }
}
```
To
```json
{
  "DummyRule": {
    "anyOf": [
      {
        "$ref": "#/definitions/AllowWarnDeny"
      },
      {
        "type": "array",
        "items": [
          {
            "$ref": "#/definitions/AllowWarnDeny"
          },
          {
            "type": "object"
          }
        ],
        "minItems": 2,
        "maxItems": 2
      }
    ]
  }
}
```

https://github.com/oxc-project/oxc/blob/87a6724f167740ca116e73bf3f611d94ec7645e0/npm/oxlint/configuration_schema.json#L217

**Comment by connorshea** [comment]
https://oxc.rs/docs/guide/usage/linter/rules/eslint/eqeqeq

Some rules use tuple shapes, for example:

```json
{
  "rules": {
    "eqeqeq": ["error", "always", { "null": "ignore" }]
  }
}
```

So we can't assume that it's always a single object.
